module/zfs/zvol.c

```diff
@@ -65,7 +65,17 @@ static list_t zvol_state_list;
 #define	ZVOL_HT_SIZE	1024
 static struct hlist_head *zvol_htable;
 #define	ZVOL_HT_HEAD(hash)	(&zvol_htable[(hash) & (ZVOL_HT_SIZE-1)])
-static DEFINE_IDA(zvol_ida);
+
+/*
+ * Note: this is an unrolled version of the DEFINE_IDA() macro,
+ * modified to not use a cast.  The cast does not compile with the
+ * compiler flags used for ZoL (-std=gnu99).
```

Comment:
Yes, that would work. But why would casting cause error? I didn't get any error when I try this patch with DEFINE_IDA.